src/fpm/fpm_release.f90

```diff
@@ -0,0 +1,46 @@
+!># Release parameters
+!> Module fpm_release contains public constants storing this build's unique version IDs
+module fpm_release
+    use fpm_versioning, only: version_t,new_version
+    use fpm_error, only: error_t, fpm_stop
+    implicit none
+    private
+
+    public :: fpm_version
+    public :: version_t
+
+    contains
+
+    !> Return the current fpm version from fpm_version_ID as a version type
+    type(version_t) function fpm_version()
+
+        type(error_t), allocatable :: error
+
+        ! Accept solution from https://stackoverflow.com/questions/31649691/stringify-macro-with-gnu-gfortran
+        ! which provides the "easiest" way to pass a macro to a string in Fortran complying with both
+        ! gfortran's "traditional" cpp and the standard cpp syntaxes
+
+#ifndef FPM_RELEASE_VERSION
+#  define FPM_RELEASE_VERSION UNDEFINED
+#endif
+
+#ifdef __GFORTRAN__ /* traditional-cpp stringification */
+#  define STRINGIFY_START(X) "&
+#  define STRINGIFY_END(X) &X"
+#else               /* default stringification */
+#  define STRINGIFY_(X) #X
+#  define STRINGIFY_START(X) &
+#  define STRINGIFY_END(X) STRINGIFY_(X)
+#endif
+
+        character (len=:), allocatable :: ver_string
+        ver_string = STRINGIFY_START(FPM_RELEASE_VERSION)
+        STRINGIFY_END(FPM_RELEASE_VERSION)
```

Comment:
Finally, gfortran needs some pretty verbose preprocessing, even just to pass along a string, because it uses the "traditional" version of the cpp preprocessor